src/datatable/datatable.jl

```diff
@@ -602,7 +594,7 @@ Base.setindex!(dt::DataTable, v, ::Colon, col_inds) =
     (dt[col_inds] = v; dt)
 
 # Special deletion assignment
-Base.setindex!(dt::DataTable, x::Void, col_ind::Int) = delete!(dt, col_ind)
+Base.setindex!(dt::DataTable, x::Null, col_ind::Int) = delete!(dt, col_ind)
```

Comment:
Yeah, that sounds like consequence of the R inspiration. Let's leave this alone in this PR and deprecate it in another one.